Integration/CityStateFromPostalCodeIntegration.php

```diff
@@ -110,8 +142,7 @@ public function doEnhancement(Lead &$lead)
 
         if (
             (empty($leadCity) || empty($leadState) || empty($leadCounty))
-            && !empty($leadCountry)
-            && !empty($leadZipCode)
+            && !(empty($leadCountry) || empty($leadZipCode))
```

Comment:
Think this is wrong:
If we don't have country at this point, then we cannot deterministically look up anything.
If we have a zipcode, city and state lookup will not happen... so we're breaking previous functionality here.

Reply:
It is not. Practical Logic 101:
NOT a AND NOT b === NOT (a OR b)

> DeMorgan's Theorem
De Morgan's theorem allows large bars in a Boolean Expression to be broken up into smaller bars over individual variables.
De Morgan's theorem says that a large bar over several variables can be broken between the variables if the sign between the variables is changed.
De Morgan's theorem can be used to prove that a NAND gate is equal to an OR gate with inverted inputs.
De Morgan's theorem can be used to prove that a NOR gate is equal to an AND gate with inverted inputs.
In order to reduce expressions with large bars, the bars must first be broken up. This means that in some cases, the first step in reducing an expression is to use De Morgan's theorem.
It is highly recommended to place parentheses around terms where lines have been broken.

Reply:
It is.

Reply:
Ok... but why on earth write it this way? Am I the only one having difficulty understanding this?